vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_client.go

```diff
@@ -114,14 +114,13 @@ func newContainerMetricsSnapshot(containerMetrics v1beta1.ContainerMetrics, podM
 }
 
 func calculateUsage(containerUsage k8sapiv1.ResourceList) model.Resources {
-	cpuQuantity := containerUsage[k8sapiv1.ResourceCPU]
-	cpuMillicores := cpuQuantity.MilliValue()
-
-	memoryQuantity := containerUsage[k8sapiv1.ResourceMemory]
-	memoryBytes := memoryQuantity.Value()
-
-	return model.Resources{
```

Comment:
I am curious what is the current behaviour when CPU or memory resources are missing? It seems many places assume both resources exist? Will this change impact other parts of the recommender where it assumes both CPU and memory resources will be present?

E.g., https://sourcegraph.com/github.com/kubernetes/autoscaler@d17b1aa/-/blob/vertical-pod-autoscaler/pkg/recommender/input/spec/spec_client.go?L118 and https://sourcegraph.com/github.com/kubernetes/autoscaler@d17b1aa/-/blob/vertical-pod-autoscaler/pkg/recommender/logic/recommender.go?L66